src/main/java/com/myproject/controller/UserController.java

```diff
@@ -0,0 +1,180 @@
+package com.myproject.controller;
+
+import com.myproject.entity.Basket;
+import com.myproject.entity.User;
+import com.myproject.entity.enums.UserRole;
+import com.myproject.service.BasketService;
+import com.myproject.service.UserService;
+import com.myproject.util.PasswordUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@Controller
+@SessionAttributes({"user", "basket"})
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private BasketService basketService;
+
+    @ModelAttribute("user")
+    public User setupUser() {
+        return new User();
+    }
+
+    @ModelAttribute("basket")
+    public Basket setupProduct() {
+        return new Basket();
+    }
+
+    @RequestMapping(value = "/", method = RequestMethod.GET)
+    public String login() {
+        return "/index";
+    }
+
+    @RequestMapping(value = "/login", method = RequestMethod.POST)
```

Comment:
better create separate controller and move all logic with authorization there

Reply:
@Sofasmile If I understand right, we will do it in future when be done Spring Security. For now, we should separate our controllers by entity.